src/rust/sqs-executor/src/redis_cache.rs

```diff
@@ -107,26 +118,109 @@ impl RedisCache {
         }
     }
 
+    async fn _get_all<CA>(
```

Comment:
It could still be useful to have a `get` method from an interface perspective, but I can see a lot of value in having its implementation be in terms of `get_all` just to reduce the amount of logic we have to write.

If we measure things and those extra CPU cycles turn out to be a bottleneck, we can revisit. Until that time, less code is better 👍